modules/labs-react/text-input/lib/hooks/useTextInputModel.tsx

```diff
@@ -0,0 +1,45 @@
+import {
+  createEventMap,
+  Model,
+  ToModelConfig,
+  useEventMap,
+  useUniqueId,
+} from '@workday/canvas-kit-react/common';
+
+type TextInputState = {
+  hasError?: boolean;
+  inputId?: string;
+  hintId?: string;
+  isRequired?: boolean;
+};
+
+type TextInputEvents = {};
+
+export type TextInputModel = Model<TextInputState, TextInputEvents>;
+
+const textInputEventMap = createEventMap<TextInputEvents>()({
+  guards: {},
+  callbacks: {},
+});
+
+export type TextInputModelConfig = TextInputState &
+  Partial<ToModelConfig<TextInputState, TextInputEvents, typeof textInputEventMap>>;
+
+export const useTextInputModel = (config: TextInputModelConfig = {}): TextInputModel => {
+  const inputId = useUniqueId(config.inputId);
+  const hintId = useUniqueId(config.hintId);
```

Comment:
Other components that have multiple ids use the same base id and derive new ones from there. `Tabs` for example has a single `id` and derives item ids from it. This decreases the amount of state necessary.